src/Syntax/TimeAgnostic.hs

```diff
@@ -50,22 +50,13 @@ import Syntax.Timing  as Syntax.TimeAgnostic
 import Syntax.Untyped as Syntax.TimeAgnostic (Id, Pn, Interface(..), EthEnv(..), Decl(..), SlotType(..), ValueType(..))
 
 -- AST post typechecking
-data Claim t
-  = C (Constructor t)
-  | B (Behaviour t)
-  | I (Invariant t)
-  | S Store
-deriving instance Show (InvariantPred t) => Show (Claim t)
-deriving instance Eq   (InvariantPred t) => Eq   (Claim t)
+data Act t = Act Store [Contract t]
+deriving instance Show (InvariantPred t) => Show (Act t)
+deriving instance Eq   (InvariantPred t) => Eq   (Act t)
 
-
--- TODO maybe enforce this structure to Act programs
--- Store * [Contract of (C, [B], [I],)]
-
-data Transition t
-  = Ctor (Constructor t)
-  | Behv (Behaviour t)
-  deriving (Show, Eq)
+data Contract t = Contract [Constructor t] [Behaviour t]
```

Comment:
why is this a list of constructors?

Reply:
Because in the typed AST we're splitting constructors into pass and fail modes, so we will have two constructors for each contract. But in a branch that builds on top of this one, I remove modes, and this is not a list anymore.